### PR TITLE
Cleanup RequestOptions

### PR DESCRIPTION
**Pull request by omus**
- Removed unused parameters from RequestOptions
- Changed RequestOptions from mutable to immutable
- Create accessors for RequestOptions

### CODE REVIEW

**Comment by omus** [comment]
@rofinn any ideas with the CI failure?

**Comment by rofinn** [comment]
Looks like the same concurrency issue. I just missed it because we're using `hex2bytes`.

File: src/FTPObject.jl [hunk 1]
```diff
@@ -47,11 +47,13 @@ mutable struct FTP
...
-    println(io, "User:      $(o.username)")
-    println(io, "Transfer:  $(o.active_mode ? "active" : "passive") mode")
-    println(io, "Security:  $(o.ssl ? (o.implicit ? "implicit" : "explicit") : "None")")
+    opts = ftp.ctxt.options
+    join(io, [
+        "Host:      $(ftp.ctxt.url)",
+        "User:      $(username(opts))",
+        "Transfer:  $(ispassive(opts) ? "passive" : "active") mode",
+        "Security:  $(security(opts))",
+    ], "\n")
```
Comment (ararslan) [inline src/FTPObject.jl:56]: Just FYI you could also do this as a single call to `print` with a triple quoted string.

**Review by ararslan — APPROVED** [review]
This is probably the third time ever in my life that I've looked at this package but these changes seem reasonable to me 😛

File: src/FTPObject.jl (hunk 1 above)
Comment (omus) [reply src/FTPObject.jl:56]: Good point!

**Comment by rofinn** [comment]
LGTM. Looks like a random/weird connection error on windows, but I'm not too concerned. If someone is using FTPClient.jl on Windows and hits this error they can make an issue.

**Pull request merged**